zerver/views/streams.py

```diff
@@ -316,16 +317,24 @@ def update_stream_backend(
             raise JsonableError(_("Invalid parameters"))
 
     if is_private is not None:
+        if is_default_stream is not None:
+            proposed_is_default_stream = is_default_stream
+        else:
+            default_stream_ids = {s.id for s in get_default_streams_for_realm(stream.realm_id)}
+            proposed_is_default_stream = stream.id in default_stream_ids
```

Comment:
I think we can move this logic to set `proposed_is_default_stream` outside this if-block for better readability. And then have a different block to decide the error message.

Reply:
I am also not sure what error message to show when stream was neither default nor private initailly and the user passes both `is_private` and `is_default_stream` as `True` in the request.